evm.go

```diff
@@ -69,6 +69,7 @@ func EvmChainIdToChainSelector() map[uint64]uint64 {
 	return copyMap
 }
 
+// Deprecated, this only supports EVM chains, use the chain agnostic `GetChainIDFromSelector` instead
```

Comment:
hm are we sure we want to direct to that and not some new EVMChainIdFromSelector which does have the uint64 typing? I think if ChainIdFromSelector was used in an EVM specific context (say in the EVM relayer) it'd be annoying to have to strconv

Reply:
in that case I guess users can still rely on the evm specific getter. That's why I think it will be important to at some point create submodules for all these methods, to make the switch explicit. 

I'm going to move forward with the merge, however I'm happy to continue this topic in slack